rpc/jsonrpc/eth_block.go

```diff
@@ -332,6 +337,46 @@ func (api *APIImpl) GetBlockByHash(ctx context.Context, numberOrHash rpc.BlockNu
 	return response, err
 }
 
+func (api *APIImpl) checkAndFillArbClassicL1BlockNumber(ctx context.Context, block *types.Block, response map[string]interface{}, chainConfig *chain.Config, tx kv.Tx) {
+	if chainConfig.IsArbitrum() && !chainConfig.IsArbitrumNitro(block.Number()) {
+		l1BlockNumber, err := api.fillArbClassicL1BlockNumber(ctx, block, tx)
+		if err != nil {
+			log.Error("error trying to fill legacy l1BlockNumber", "err", err)
+		} else {
+			response["l1BlockNumber"] = l1BlockNumber
+		}
+	}
+}
+
+func (api *APIImpl) fillArbClassicL1BlockNumber(ctx context.Context, block *types.Block, tx kv.Tx) (hexutil.Uint64, error) {
+	startBlockNum := block.Number().Int64()
+	blockNum := startBlockNum
+	i := int64(0)
+	for {
+		transactions := block.Transactions()
+		if len(transactions) > 0 {
+			legacyTx, ok := transactions[0].(*types.ArbitrumLegacyTxData)
```

Comment:
assertion here always happen? we do not have classic arbitrum support yet so cant really test, please keep reference to original code in comment here so we can easily get back to it when revisit code for classic support